pkg/cmd/subcommand/demo.go

```diff
@@ -0,0 +1,270 @@
+	availableVersions = []string{"v0.3.1", "v0.4.0", "latest"}
+}
+
+// NewDemo returns an initialized Init instance
+func NewDemo(ioStreams genericclioptions.IOStreams) *Demo {
+	return &Demo{
+		IOStreams: ioStreams,
+	}
+}
+
+func NewCmdDemo(restClient util.Getter, ioStreams genericclioptions.IOStreams) *cobra.Command {
+	var cl *k8s.Clientset
+
+	i := NewDemo(ioStreams)
+
+	cmd := &cobra.Command{
+		Use:                   "demo <args>...",
+		DisableFlagsInUseLine: true,
+		Short:                 "Create OpenFunction demo.",
+		Long: `
+This command will help you to Run the OpenFunction demo.
+
+You can use fn demo  to run the OpenFunction demo.
+
+The demo running is OpenFunction v0.4.0 are: Dapr, KEDA, Knative Serving, Shipwright, Tekton Pipelines.
+
+The permitted parameters are: --delete, --region-cn, --version, --verbose, --dry-run
+
+The dependencies to be installed for OpenFunction v0.4.0 are: Dapr, KEDA, Knative Serving, Shipwright, Tekton Pipelines, Cert Manager, Ingress Nginx.
+
+The permitted parameters are: --delete, --region-cn, --version, --verbose, --dry-run
+`,
+		Example: "fn demo",
+		Run: func(cmd *cobra.Command, args []string) {
+			if i.Delete {
+				util.CheckErr(i.DeleteKind(cmd, args))
+			} else {
+				util.CheckErr(i.RunKind(cl, cmd))
+				installCMD := NewCmdInstall(restClient, ioStreams)
+
+				installCMD.SetArgs([]string{
+					fmt.Sprintf("install"),
+					fmt.Sprintf("--all"),
+					fmt.Sprintf("--region-cn=%v", i.RegionCN),
+					fmt.Sprintf("--version=latest")},
+				)
+				installCMD.Execute()
+				util.CheckErr(i.RunDemo(cl, cmd))
+			}
+		},
+	}
+
+	cmd.Flags().BoolVar(&i.Verbose, "verbose", false, "Show verbose information.")
+	cmd.Flags().BoolVar(&i.RegionCN, "region-cn", false, "For users in China to uninstall dependent components.")
+	cmd.Flags().BoolVar(&i.DryRun, "dry-run", false, "Used to prompt for the components and their versions to be uninstalled by the current command.")
+	cmd.Flags().BoolVar(&i.Delete, "delete", false, "Used to specify the version of OpenFunction to be uninstalled.")
+	cmd.Flags().DurationVar(&i.Timeout, "timeout", 5*time.Minute, "Set timeout time. Default is 5 minutes.")
+	// In order to avoid too many options causing misunderstandings among users,
+	// we have hidden the following parameters,
+	// but you can still find their usage instructions in the documentation.
+	cmd.Flags().MarkHidden( "version")
+	cmd.Flags().MarkHidden("ingress")
+	cmd.Flags().MarkHidden("cert-manager")
+	cmd.Flags().MarkHidden("shipwright")
+	cmd.Flags().MarkHidden("keda")
+	cmd.Flags().MarkHidden("dapr")
+	return cmd
+}
+
+
+func (i *Demo) RunKind(cl *k8s.Clientset, cmd *cobra.Command) error {
+	operator := common.NewOperator(runtime.GOOS, i.OpenFunctionVersion, i.Timeout, i.RegionCN, i.Verbose)
+	ti := util.NewTaskInformer("Demo")
+	continueFunc := func() bool {
+		reader := bufio.NewReader(os.Stdin)
+		fmt.Fprintln(w, ti.BeforeTask("We will create a Kind cluster and run the OpenFunction Demo..\n"+
```

Comment:
We will create a Kind cluster and run the OpenFunction Demo..
=>
A Kind cluster will be created in which an OpenFunction Demo will be launched.